### PR TITLE
DINT-29: expand collector to forward to AEP through alloy

### PR DESCRIPTION
**Pull request by aniham**
changes:

- create AEP ctx, set default ctx (empty)
- create EventForwarding ctx, set default ctx (`snowplow: true, aep: false`)
- define schemas for new contexts
- initialize alloy (AEP tracker) with AEP context
- send pageView events (correct xdm schema) - first deliverable

### CODE REVIEW

File: src/types/index.d.ts [hunk 1]
```diff
@@ -11,5 +11,6 @@ declare global {
 
     interface Window {
         magentoStorefrontEvents: MagentoStorefrontEvents;
+        alloy: any;
```
Comment (abrhim) [inline src/types/index.d.ts:14]: Does Alloy have a type we can assign here?

**Review by arjunathreya141291 — APPROVED** [review]

File: src/handlers/shoppingCart/viewAEP.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+import { Event } from "@adobe/magento-storefront-events-sdk/dist/types/types/events";
+/* eslint-disable */
+const aepHandler = (event: Event): void => {
+    // TODO stub
```
Comment (evargast) [inline src/handlers/shoppingCart/viewAEP.ts:9]: very minor: it's usually good practice to add a `console.error("aepHandler has not been implemented")` for empty methods that have yet to be implemented but are already exported

File: src/handlers/shoppingCart/view.ts [hunk 3]
```diff
@@ -4,29 +4,23 @@
...
-import {
-    SelfDescribingJson,
-    trackStructEvent,
-} from "@snowplow/browser-tracker";
 
-import { createShoppingCartCtx } from "../../contexts";
+import { createEventForwardingCtx } from "../../contexts";
+import { EventForwardingContext } from "../../types/contexts";
+import aepHandler from "./viewAEP";
+import snowplowHandler from "./viewSnowplow";
```
Comment (evargast) [inline src/handlers/shoppingCart/view.ts:11]: What are your thoughts on the generic handler names? I find it kinda confusing but not sure if it improves your dev experience

**Review by evargast — APPROVED** [review]
Changes look good, it would be great if we could add some more tests to reliably test these features... is there another ticket to write tests for these features?

File: src/types/index.d.ts (hunk 1 above)
Comment (aniham) [reply src/types/index.d.ts:14]: no, tragically as @evargast would say - i put in a request in the aep-web-sdk channel and they suggested i add it to alloy feature requests directly - will do:

https://github.com/adobe/alloy/projects/6

File: src/types/@adobe.alloy.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+declare module "@adobe/alloy";
```
[the remark this answers is not in the document]
Comment (aniham) [reply src/types/@adobe.alloy.d.ts:1]: addressed under abram's comment but adding here too - will submit:

https://github.com/adobe/alloy/projects/6

File: src/handlers/shoppingCart/viewSnowplow.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+import { Event } from "@adobe/magento-storefront-events-sdk/dist/types/types/events";
```
[the remark this answers is not in the document]
Comment (aniham) [reply src/handlers/shoppingCart/viewSnowplow.ts:6]: we can add to the notes in [DINT-97](https://jira.corp.magento.com/browse/DINT-97) - i'd appreciate a chime-in with a suggestion on renames :)

File: src/handlers/shoppingCart/viewAEP.ts (hunk 2 above)
Comment (aniham) [reply src/handlers/shoppingCart/viewAEP.ts:9]: done

File: src/handlers/shoppingCart/view.ts (hunk 3 above)
Comment (aniham) [reply src/handlers/shoppingCart/view.ts:11]: you mean `/page/view.ts` vs `pageView.ts`? i'm in favor of the latter but i tried to be as as undisruptive as possible - can be another thing to address in DINT-97

**Pull request merged**